stock_lot_analytic/models/stock_lot.py

```diff
@@ -1,9 +1,29 @@
 # Copyright 2023 Quartile Limited
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import models
+from odoo import _, models
 
 
 class StockLot(models.Model):
     _name = "stock.lot"
     _inherit = ["stock.lot", "analytic.mixin"]
+
+    def write(self, vals):
+        if "analytic_distribution" not in vals:
+            return super().write(vals)
+        new_dist = vals.get("analytic_distribution", {})
+        new_formatted = "<br/>".join(self._format_distribution(new_dist))
+        for record in self:
+            old_dist = record.analytic_distribution or {}
+            if old_dist == new_dist:
+                continue
+            old_formatted = "<br/>".join(self._format_distribution(old_dist))
+            record.message_post(
+                body=_(
+                    "Analytic Distribution updated<br/><br/>"
+                    "To:<br/> %(to)s<br/><br/>"
+                    "From:<br/> %(from)s"
+                )
+                % {"to": new_formatted, "from": old_formatted}
+            )
+        return super(StockLot, self).write(vals)
```

Comment:
Non-blocking - To stick the practice.
```suggestion
        return super().write(vals)
```

Reply:
Done